docs/migration/manual-migration.mdx

```diff
@@ -11,6 +11,24 @@ This manual migration process should be run after the
 [automated migration process](./automated-migration), to complete the missing
 parts, or debug issues in the migration CLI output.
 
+## Transitioning to ESM
+
+Starting 3.0, in a effort to modernize the codebase, we transitioned to ESM. You
```

Comment:
**nitpick:** it may not be clear for readers what ESM is, and this doc doesn't explain it

**suggestion:** add a sentence explaining it, and ideally one or more link to external resources